### ISSUE TITLE
Inputs size are not applied inside Space.Compact component

### ISSUE THREAD

**Issue by vitalijalbu** [ant-design/ant-design#42724]
### Reproduction link

[https://ant.design/components/form](https://ant.design/components/form)

### Steps to reproduce

```
<Form layout="vertical">
  <Space>
    <Space.Compact block>
      <AddressField size="large" />
      <Form.Item name="category" label="Seleziona categoria">
        <Select
          size="large"
        />
      </Form.Item>
      <Form.Item name="type" label="Seleziona categoria">
        <Select
          size="large"
        />
      </Form.Item>
    </Space.Compact>
    <Button
      size="large"
    >
      Ricerca
    </Button>
  </Space>
</Form>
```

### What is expected?

Need to make select and input large

### What is actually happening?

Inputs are not large, if removing the space compact, all is ok.

| Environment | Info |
| --- | --- |
| antd | 5.5.2 |
| React | 18 |
| System | macos |
| Browser | chrome |

**Comment by vitalijalbu** [comment]
@zombieJ 
https://codesandbox.io/s/sleepy-darkness-9e5toh?file=/src/App.js

**Comment by daryazata** [comment]
hi, i would like to work on this issue (butthis  will be my first issue)

**Comment by daryazata** [comment]
@linercoder  just wondering, wasnt it already solved with your pr https://github.com/ant-design/ant-design/pull/42752

**Comment by linxianxi** [comment]
> @linxianxi just wondering, wasnt it already solved with your pr #42752

yes

**Comment by Wxh16144** [comment]
antd 4.x 也有这个问题，还修不修啊
